07.ls_object/ls.rb

```diff
@@ -14,35 +14,26 @@ def main
   opt.on('-l') { options[:long_format] = true }
 
   paths = opt.parse(ARGV)
+  options[:header] = !(paths.length == 1 || paths.empty?)
   paths << '.' if paths.empty?
 
-  stats = LsFileStat.bulk_create(paths, reverse: options[:reverse])
-  non_recursive_stats = stats.reject(&:directory?)
-  recursive_stats = stats.select(&:directory?)
+  screen_src = { '' => [] }
 
-  non_recursive_result = Screen.new(non_recursive_stats, options).show
-  # 再帰処理の結果にディレクトリ名のヘッダを表示するかどうかを制御
-  options[:header] = true if !non_recursive_result.nil? || recursive_stats.length > 1
-  recursive_results = create_recursive_results(recursive_stats, options)
-
-  puts [non_recursive_result, recursive_results].compact.join("\n\n")
-end
-
-def create_recursive_results(stats, options)
-  return nil if stats.empty?
-
-  stats.map do |stat|
-    globed_files = Dir.glob('*', (options[:all_visible] ? File::FNM_DOTMATCH : 0), base: stat.name)
-    globed_files << '..' if options[:all_visible]
-
-    result = ''
-    Dir.chdir(stat.name) do
-      stats_in_dir = LsFileStat.bulk_create(globed_files, reverse: options[:reverse])
-      result = Screen.new(stats_in_dir, options).recursive_show(base: stat.name)
+  paths.each do |path|
+    stat = LsFileStat.new(path)
+    if stat.file?
+      screen_src[''] << stat
+    else
+      screen_src[path] = create_stats_in_directory(path, options)
     end
-
-    result
   end
+  Screen.new(screen_src, options).show
+end
+
+def create_stats_in_directory(base_path, options)
+  globed_files = Dir.glob('*', (options[:all_visible] ? File::FNM_DOTMATCH : 0), base: base_path)
```

Comment:
```suggestion
  files = Dir.glob('*', (options[:all_visible] ? File::FNM_DOTMATCH : 0), base: base_path)
```

でいいかも？

https://bootcamp.fjord.jp/pages/avoid-too-specific-var-name